packages/kit/src/runtime/client/renderer.js

```diff
@@ -384,6 +384,16 @@ export class Renderer {
 					deep_linked.scrollIntoView();
 				} else {
 					scrollTo(0, 0);
+					if (document.body.scrollTop > 0) document.body.scrollTo(0, 0);
+					// In some cases it is needed scroll first root element of target that
+					// is not scrolled to the top
```

Comment:
```suggestion
					// If there is an element with the overflow CSS property it may have a scrollbar
					// We need to ensure the scroll is reset on these elements
```